test/mpi/main.cc

```diff
@@ -383,34 +397,103 @@ void ProcessArgs(int argc, char** argv)
     }
 }
 
+#ifdef HAVE_CUDA
+void cuda_set_device()
+{
+    MPI_Comm local_comm;
+    int cuda_dev_count;
+    int local_rank;
+    MPI_Comm_split_type(MPI_COMM_WORLD, OMPI_COMM_TYPE_NODE, 0, MPI_INFO_NULL,
```

Comment:
this is OMPI specific. Is there an MPI-spec way to do it? (so that it works e.g. with MPICH). We use MPI only to bootstrap ranks here, so any should work imho.

Reply:
Should be OK with MPICH as well: https://www.mpich.org/static/docs/v3.2/www3/MPI_Comm_split_type.html